services/src/main/java/org/keycloak/services/filters/CachingGZIPEncodingInterceptor.java

```diff
@@ -0,0 +1,90 @@
+package org.keycloak.services.filters;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.output.TeeOutputStream;
+import org.jboss.resteasy.plugins.interceptors.encoding.GZIPEncodingInterceptor;
+import org.jboss.resteasy.resteasy_jaxrs.i18n.LogMessages;
+import org.keycloak.common.Profile;
+
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.Provider;
+import javax.ws.rs.ext.WriterInterceptorContext;
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Provider
+@Priority(Priorities.ENTITY_CODER)
+public class CachingGZIPEncodingInterceptor extends GZIPEncodingInterceptor {
+
+    private final Path tmpDir;
+    @Context
+    private UriInfo uriInfo;
+
+    public CachingGZIPEncodingInterceptor() {
+        try {
+            this.tmpDir = Files.createTempDirectory("gzip-cache");
+        } catch (IOException e) {
+            throw new RuntimeException("could not create temp gzip cache directory");
```

Comment:
I would go for a warning here and state something like gzip compression is disabled, rather than fail.